src/noisepy/seis/application_modules/esyn_utils.py

```diff
@@ -0,0 +1,149 @@
+import math
+
+import numpy as np
+import pyasdf
+
+### -----
+# This scripts are aim to perform the 2-D radiative transfer equation
+# for scalar waves (Shang and Gao 1988; Sato 1993),
+# assuming of isotropic scattering and source radiation in infinite medium
+# to calculate synthesized energy densities Esyn.
+### -----
+
+
+### -----
+def read_pyasdf(sfile, ccomp):
+    # useful parameters from each asdf file
+    with pyasdf.ASDFDataSet(sfile, mode="r") as ds:
+        alist = ds.auxiliary_data.list()
+        try:
+            dt = ds.auxiliary_data[alist[0]][ccomp].parameters["dt"]
```

Comment:
I wonder if the StackStore could be used for this instead of direct asdf file access? That way your function would work for any of the supported formats (asdf, zarr, numpy)

Reply:
It is a good point that using StackStore makes the function more practical. While the applications are all still currently relying on ASDF, maybe we can leave this to the next PR. What do you think?

Reply:
Yeah, there can a subsequent PR to update all of them

Reply:
Can you create an issue for this? It could be a good CodeUW task

Reply:
Issue created.